### ISSUE TITLE
Public key auth not working

### ISSUE THREAD

**Issue by TraGicCode** [DarwinJS/ChocoPackages#21]
I am unable to get public key authentication working when installing via the choco package.  I simply keep getting access denied.  What i did notice is if i run the C:/Program Files/OpenSSH-Win64/install-sshlsa.ps1 script manually all seems to work perfect.  Any ideas on what's going on?  I don't need anything of failure or warning in the choco logs

Here is the command i'm using

choco install openssh -params "/SSHServerFeature"

**Comment by TraGicCode** [comment]
The following older version seems to work fine using the below command

choco install -y win32-openssh -version 2016.05.30.20160902 --params='/SSHServerFeature /KeyBasedAuthenticationFeature /UseNTRights'

unfortunately the new version doesn't.

**Comment by felfert** [comment]
I just stumbled over the same issue. Reading the code in chocolateyinstall.ps1 immediately revealed the reason. ssh-lsa.dll gets installed only, if it is **already existing** _and_ **has changed in size**. of course, that is completely wrong. The fix is trivial. I have created PR #25 for that. Hopefully it gets merged soon.

Until then, you can use this simple workaround **before** running choco install openssh:
````
$lsadll = "$env:windir\system32\ssh-lsa.dll"
if (!(Test-Path $lsadll)) {
    New-Item -type file -path $lsadll -force | Out-Null
}
````
-Fritz

PS: Wondering, why not more people have come across this. The package hat 39k+ downloads?!

**Comment by DarwinJS** [comment]
@felfert - thanks very much for spotting this!

My original intent was one variable for both non-existence and upgrade (and the one variable is used in other places in the code) - so I renamed the existing variable more clearly and added the else statement fix you aptly pointed out!

Version 0.0.9.20170306 is in automated moderation.

**Comment by felfert** [comment]
Great! So I'm going to close the PR and hope for a release soon :-)

Cheers
 -Fritz

**Comment by DarwinJS** [comment]
If you pull it by version you can do so now.

`cinst openssh -version 0.0.9.20170306 -y`

Thanks for taking a deeper look and reporting - sometimes it can be hard to sort through all the messages on this project and understand what are actually problems with the choco package - appreciate the second set of eyes!

**Comment by felfert** [comment]
Hmm, after reading your changed code, is still like my variant with two vars better. Reason:
The following code prints a false warning, even if the dll was newly installed:
````
If ($SSHLsaNeedsUpdating)
  {
    Write-Warning "IMPORTANT: You must reboot so that key based authentication can be fully installed or upgraded for the SSHD Service."
}
````

**Comment by DarwinJS** [comment]
As far as I am aware you must reboot to get the new DLL to be part of Lsa's enabled authentication methods even on the first install because it only reads them at boot up.

**Comment by felfert** [comment]
Ok, you are probably right.
I'm using choco from inside a cloudbase-init script on a freshly generated W2012R2 on OpenStack. So the VM is shutdown and snapshotted at the end anyway.

**Comment by DarwinJS** [comment]
@felfert - There was a knock on error with the code that copies ssh-lsa.dll into the system32 folder on a fresh install.

It is fixed in version 0.0.9.20170308, which as of this moment is still in moderation.  It can be pulled before moderation is complete by specifying the version on your chocolately install or upgrade command line.
